.github/workflows/release.yaml

```diff
@@ -0,0 +1,40 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: release
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/project/taegis-magic/
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.8'
+    - name: Install dependencies
+      run: python -m pip install --upgrade pip build hatch
+    - name: Build package
+      run: python -m hatch build
+    - name: Publish package distributions to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
Overall this PR looks good.

A question or two though. While I'm not a GH actions/workflow expert, it looks like the build and publish steps are in the same job, instead of separate jobs. We'd like them in separate jobs with id-token: write only on the publish correct? And perhaps have builds be done on any merge but publish only manually or on tags?

Reply:
We will need to investigate how we would scan the build and what solutions are available.  Until we have check for the build, separating the build from the publish won't have any effect other than introducing more complexity.